pkg/rc/auditing_transaction.go

```diff
@@ -0,0 +1,95 @@
+package rc
+
+import (
+	"context"
+
+	"github.com/square/p2/pkg/audit"
+	pc_fields "github.com/square/p2/pkg/pc/fields"
+	"github.com/square/p2/pkg/store/consul/transaction"
+	"github.com/square/p2/pkg/types"
+	"github.com/square/p2/pkg/util"
+)
+
+type auditingTransaction struct {
+	ctx           context.Context
+	nodes         map[types.NodeName]struct{}
```

Comment:
I could have done `bool`, but `struct{}` occupies 0 bytes of memory so it's slightly more "efficient" even though it doesn't really matter here.

source: https://dave.cheney.net/2014/03/25/the-empty-struct